src/function/vector_string_functions.cpp

```diff
@@ -252,16 +255,47 @@ function_set RegexpMatchesFunction::getFunctionSet() {
     return functionSet;
 }
 
+std::unique_ptr<FunctionBindData> regexReplaceBindFunc(ScalarBindFuncInput input) {
+    RegexpReplaceFunction::RegexReplaceOption replaceOption =
+        RegexpReplaceFunction::RegexReplaceOption::FIRST_OCCUR;
+    if (input.arguments.size() == 4) {
+        auto option = input.arguments[3];
+        if (option->expressionType != ExpressionType::LITERAL ||
+            option->getDataType() != LogicalType::STRING()) {
+            throw common::BinderException{"option of regex_replace must be a constant string."};
```

Comment:
Cover. Also try to use ExpressionUtils to validate data type & get literal values. There are helper functions.